addons/binding/org.openhab.binding.irtrans/src/main/java/org/openhab/binding/irtrans/handler/EthernetBridgeHandler.java

```diff
@@ -0,0 +1,925 @@
+                    disconnect(socketChannel);
+                }
+            }
+        } else {
+            logger.warn("Did not receive an answer from the IRtrans transceiver for '{}' - Parsing is skipped",
+                    getThing().getUID());
+            disconnect(socketChannel);
+        }
+
+        return commandList;
+    }
+
+    private String[] getRemoteList(int index) {
+        String getRemotes = "Agetremotes " + index + (char) 13;
+        ByteBuffer response = sendCommand(getRemotes);
+        String[] remoteList = new String[0];
+
+        if (response != null) {
+            String message = stripByteCount(response).split("\0")[0];
+            logger.trace("remotes returned {}", message);
+            if (message != null) {
+                if (message.contains("REMOTELIST")) {
+                    remoteList = message.split(",");
+                } else {
+                    logger.warn("Received some non-compliant command ({})", message);
+                    disconnect(socketChannel);
+                }
+            }
+        } else {
+            logger.warn("Did not receive an answer from the IRtrans transceiver for '{}' - Parsing is skipped",
+                    getThing().getUID());
+            disconnect(socketChannel);
+        }
+
+        return remoteList;
+    }
+
+    public boolean sendIRcommand(IrCommand command, Led led) {
+        // construct the string we need to send to the IRtrans device
+        String output = packIRDBCommand(led, command);
+
+        lock.lock();
+        try {
+            ByteBuffer response = sendCommand(output);
+
+            if (response != null) {
+                String message = stripByteCount(response).split("\0")[0];
+                if (message != null && message.contains("RESULT OK")) {
+                    return true;
+                } else {
+                    logger.debug("Received an unexpected response from the IRtrans transceiver: '{}'", message);
+                    return false;
+                }
+            }
+        } finally {
+            lock.unlock();
+        }
+
+        return false;
+    }
+
+    private ByteBuffer sendCommand(String command) {
+        if (command != null) {
+            ByteBuffer byteBuffer = ByteBuffer.allocate(command.getBytes().length);
+            try {
+                byteBuffer.put(command.getBytes("ASCII"));
+                onWritable(byteBuffer, socketChannel);
+                Thread.sleep(((BigDecimal) getConfig().get(RESPONSE_TIME_OUT)).intValue());
+                return onReadable(socketChannel, ((BigDecimal) getConfig().get(BUFFER_SIZE)).intValue());
+            } catch (UnsupportedEncodingException | NumberFormatException | InterruptedException e) {
+                Thread.currentThread().interrupt();
+                logger.error("An exception occurred while configurting the IRtrans transceiver for '{}': {}",
```

Comment:
Is this a bug situation that the user should report? If so, better include the full exception with stacktrace.

Reply:
typo: configurting